### PR TITLE
feat: incorporate chakra type generation in the workflow

### PR DESCRIPTION
**Pull request by Averethel**
References: https://smg-au.atlassian.net/browse/FED-421

Chakra-ui provides [type generation](https://www.chakra-ui.com/docs/theming/overview#typescript) based on the tokens defined in the theme. 

This PR integrates the typegen in our workflow and ensures that we have proper type definitions in our component library

### CODE REVIEW

File: src-v2/components/vehicleReference/index.stories.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const meta: Meta<typeof VehicleReference> = {
   component: VehicleReference,
   decorators: [
     (Story) => (
-      <Box maxW={{ '2xs': '100%', md: '600px' }}>
+      <Box maxW={{ '2xs': 'full', md: '7xl' }}>
```
Comment (Averethel) [inline src-v2/components/vehicleReference/index.stories.tsx:14]: Strict tokens mean we need to use design tokens here

File: package.json [hunk 2]
```diff
@@ -43,8 +43,11 @@
     "format:i18n": "npx sort-phraseapp-locales src/locales/de/dict.json src/locales/en/dict.json src/locales/fr/dict.json src/locales/it/dict.json --fix",
     "format": "npm run format:i18n && npm run lint -- --fix",
     "typecheck": "tsc --noEmit",
+    "typegen": "npx chakra typegen src/themes/autoScout24.ts --clean",
```
Comment (Averethel) [inline package.json:46]: We need to pass a single system; both shapes are the same.

File: src/components/icons/HeartIcon.tsx [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import React from 'react';
-import { createIcon } from '@chakra-ui/react';
+import { createIcon, IconProps } from '@chakra-ui/react';
+
+type HeartProps = IconProps & {
+  fill?: 'none';
+  stroke?: 'currentColor';
```
Comment (Averethel) [inline src/components/icons/HeartIcon.tsx:6]: extending props as these values are not supported and we need them for animations in listings

File: src/components/icons/CompareIcon.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const CompareIcon = createIcon({
   ),
   defaultProps: {
     boxSize: 'sm',
-    fill: 'currentColor',
+    fill: 'currentBg',
```
Comment (Averethel) [inline src/components/icons/CompareIcon.tsx:18]: `currentColor` is not supported, `currentBg` makes sense for `fill`

File: src/components/icons/RocketIcon.tsx [hunk 5]
```diff
@@ -13,13 +13,13 @@ export const RocketIcon = createIcon({
...
+        stroke="currentColor"
+        fill="none"
         d="M18.758 1.671a.607.607 0 0 0-.429-.428C17.197 1 16.311 1 15.429 1c-3.63 0-5.806 1.94-7.431 4.5H4.334c-.575 0-1.25.417-1.508.931L1.088 9.905a.998.998 0 0 0-.088.376.844.844 0 0 0 .844.844h3.65l-.79.79c-.4.4-.457 1.134 0 1.59l1.789 1.79c.392.394 1.13.463 1.592 0l.79-.79v3.651a.844.844 0 0 0 .844.844c.13-.005.258-.035.376-.088l3.471-1.737c.514-.256.932-.931.932-1.507v-3.671C17.05 10.369 19 8.186 19 4.576c.004-.887.004-1.773-.24-2.905Z"
       />
     </>
   ),
   defaultProps: {
     boxSize: 'sm',
-    fill: 'none',
-    stroke: 'currentColor',
```
Comment (Averethel) [inline src/components/icons/RocketIcon.tsx:23]: We don't seem to pass these as props, moving them to the path definition

File: src/components/icons/StarIcon.tsx [hunk 6]
```diff
@@ -1,5 +1,10 @@
 import React from 'react';
-import { createIcon } from '@chakra-ui/react';
+import { createIcon, IconProps } from '@chakra-ui/react';
+
+type StarProps = IconProps & {
+  fill?: 'none';
+  stroke?: 'currentColor';
```
Comment (Averethel) [inline src/components/icons/StarIcon.tsx:6]: extending props as these values are not supported and we need them for animations in listings

File: src/components/themeProvider/index.stories.tsx [hunk 7]
```diff
@@ -51,19 +51,19 @@ const meta: Meta<typeof ThemeProvider> = {
         box: (
           <Box
             display="flex"
-            direction="row"
+            flexDirection="row"
```
Comment (Averethel) [inline src/components/themeProvider/index.stories.tsx:54]: `direction` works on `Stack`, but `css` direction attribute has nothing with flex directions.

File: src/themes/shared/tokens/sizes.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { defineTokens } from '@chakra-ui/react';
 
 export const sizes = defineTokens.sizes({
+  auto: { value: 'auto' },
```
Comment (Averethel) [inline src/themes/shared/tokens/sizes.ts:4]: Adding this as a utility

File: src/themes/shared/tokens/spacing.ts [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import { defineTokens } from '@chakra-ui/react';
 
 export const spacing = defineTokens.spacing({
+  '0': { value: '0' },
```
Comment (Averethel) [inline src/themes/shared/tokens/spacing.ts:4]: Adding this as a utility

File: src/themes/stories/TokenShowcase.tsx [hunk 10]
```diff
@@ -1,27 +1,32 @@
...
 type TokenGroupName = keyof NonNullable<
   NonNullable<SystemConfig['theme']>['tokens']
 >;
 
-type TokenShowcaseProps = {
+type TokenShowcaseProps<Token extends keyof Tokens> = {
   tokenPath: string;
   renderValue?: boolean;
-  renderDemo?: (tokenValue: string) => React.ReactNode;
+  renderDemo?: (tokenName: Tokens[Token]) => React.ReactNode;
```
Comment (Averethel) [inline src/themes/stories/TokenShowcase.tsx:16]: If we pass `string` and want to use it as a token value, strict tokens would yell at us.
This is why we add type parameters so we can properly cast a token name to an expected token.

File: src/themes/stories/tokens.stories.tsx [hunk 11]
```diff
@@ -22,7 +22,7 @@ const conversionToPixels = {
 
 export const BorderRadius: StoryType = {
   render: () => (
-    <TokenShowcase
+    <TokenShowcase<'radii'>
```
Comment (Averethel) [inline src/themes/stories/tokens.stories.tsx:25]: Passing token type

**Review by marko-markovic-smg — APPROVED** [review]

File: src/themes/shared/tokens/colors.ts [hunk 12]
```diff
@@ -136,4 +136,6 @@ export const colors = defineTokens.colors({
     800: { value: '#553676' },
     900: { value: '#402859' },
   },
+  none: { value: 'none' },
+  currentColor: { value: 'currentColor' },
```
Comment (Averethel) [inline src/themes/shared/tokens/colors.ts:140]: Adding this two utilities to streamline svg icons

**Pull request merged**